code/modules/reagents/chemistry/machinery/smoke_machine.dm

```diff
@@ -33,88 +30,145 @@
-		reagents.clear_reagents()
+
+	//new efficiency from capacitors
 	efficiency = 18
 	for(var/datum/stock_part/capacitor/capacitor in component_parts)
 		efficiency += 2 * capacitor.tier
+
+	//new maximum range from servos
 	max_range = 1
 	for(var/datum/stock_part/servo/servo in component_parts)
 		max_range += servo.tier
 	max_range = max(3, max_range)
 
-/obj/machinery/smoke_machine/on_set_is_operational(old_value)
-	if(old_value) //Turned off
-		end_processing()
-	else //Turned on
-		begin_processing()
+/obj/machinery/smoke_machine/item_interaction(mob/living/user, obj/item/tool, list/modifiers)
+	. = NONE
+	if(user.combat_mode || tool.item_flags & ABSTRACT || tool.flags_1 & HOLOGRAM_1 || !user.can_perform_action(src, ALLOW_SILICON_REACH))
+		return ITEM_INTERACT_SKIP_TO_ATTACK
 
+	//transfer reagents from an open container into machine
+	if(is_reagent_container(tool) && tool.is_open_container())
+		var/obj/item/reagent_containers/RC = tool
+		var/units = RC.reagents.trans_to(src, RC.amount_per_transfer_from_this, transferred_by = user)
+		if(units)
+			to_chat(user, span_notice("You transfer [units] units of the solution to [src]."))
+			return ITEM_INTERACT_SUCCESS
+		return ITEM_INTERACT_BLOCKING
 
-/obj/machinery/smoke_machine/process()
-	if(reagents.total_volume == 0)
-		on = FALSE
-		update_appearance()
+/obj/machinery/smoke_machine/wrench_act(mob/living/user, obj/item/tool)
+	. = ITEM_INTERACT_BLOCKING
+	if(on)
+		balloon_alert(user, "turn off first!")
 		return
-	var/turf/location = get_turf(src)
-	var/smoke_test = locate(/obj/effect/particle_effect/fluid/smoke) in location
-	if(on && !smoke_test)
-		update_appearance()
-		var/datum/effect_system/fluid_spread/smoke/chem/smoke_machine/smoke = new()
-		smoke.set_up(setting * 3, holder = src, location = location, carry = reagents, efficiency = efficiency)
-		smoke.start()
-		use_energy(active_power_usage)
 
-/obj/machinery/smoke_machine/wrench_act(mob/living/user, obj/item/tool)
-	. = ..()
-	if(default_unfasten_wrench(user, tool, time = 4 SECONDS))
-		on = FALSE
+	if(default_unfasten_wrench(user, tool, time = 4 SECONDS) == SUCCESSFUL_UNFASTEN)
 		return ITEM_INTERACT_SUCCESS
-	return FALSE
 
-/obj/machinery/smoke_machine/attackby(obj/item/I, mob/user, params)
-	add_fingerprint(user)
-	if(is_reagent_container(I) && I.is_open_container())
-		var/obj/item/reagent_containers/RC = I
-		var/units = RC.reagents.trans_to(src, RC.amount_per_transfer_from_this, transferred_by = user)
-		if(units)
-			to_chat(user, span_notice("You transfer [units] units of the solution to [src]."))
-			return
-	if(default_deconstruction_screwdriver(user, "smoke0-o", "smoke0", I))
+/obj/machinery/smoke_machine/screwdriver_act(mob/living/user, obj/item/tool)
+	. = ITEM_INTERACT_BLOCKING
+	if(on)
+		balloon_alert(user, "turn off first!")
 		return
-	if(default_deconstruction_crowbar(I))
+
+	if(default_deconstruction_screwdriver(user, "smoke0", "smoke0", tool))
```

Comment:
```suggestion
	if(default_deconstruction_screwdriver(user, "smoke0-o", "smoke0", tool))
```
or am I wrong?

Reply:
Not wrong, just not nessassary because we call update appearance in the next line which corrects the icon states for us anyway.

But there's no harm so sure